README.md

```diff
@@ -1,2 +1,221 @@
+
+# Step 2: Build the Docker image
+docker build -t ghcr.io/sage-bionetworks-workflows/py-optimize-nextflow:latest .
+
+# Step 3: Push the Docker image to GHCR
+docker push ghcr.io/sage-bionetworks-workflows/py-optimize-nextflow:latest
+```
+
+
+# Reducing Compute Costs for Nextflow Pipelines on AWS guidelines
+
+Developing Nextflow pipelines for scientific research and software engineering can be computationally intensive and expensive, especially when running on cloud platforms like AWS. However, there are several strategies you can employ to optimize costs without sacrificing performance.
+
+This article is also avaiable at <https://sagebionetworks.jira.com/wiki/spaces/WF/pages/3518496775/Controlling+Execution+Costs>.
+
+### Utilize Spot Compute Instances
+
+Spot instances are a cost-effective solution for running Nextflow pipelines. These instances can be up to 90% cheaper than on-demand instances. However, there's a tradeoff: spot instances are typically reclaimed after 6-8 hours, which can interrupt long-running tasks. 
+
+**Best Practice:** Use spot instances for short and intermediate steps within your workflow. If a step in your workflow exceeds the typical spot instance duration (6-8 hours), it's better to use on-demand instances for that step. This ensures that you won't lose progress and incur additional costs from having to rerun long tasks.
+
+##### Override the compute environment on a per-step basis
+If no process queue is defined for a task it will use the default that is applied to
+the pipeline. However, there are some cases where you might want to do the following:
+
+1) Run most tasks in a spot instance
+2) Run a few long running, or otherwise "mission" critical tasks in an on-demand instance.
+3) Or the opposite where the default is an on-demand instance, or some tasks are spot
+
+This can be accomplished by defining a `queue` attribute in the `.config` file.
+
+
+In the following example I am setting a specific process to use a queue that I've 
+manually defined. For those processes that don't have a queue manually defined it will
+use the default queue defined for that pipeline.
+
+```
+process {
+  withName: my_task_name {
+      queue = 'TowerForge-queue-id'
+
+      maxErrors     = '-1'
+      maxRetries    = 3
+      errorStrategy = { task.attempt <= 3 ? 'retry' : 'finish' }
+
+      cpus   = 2
+      memory = 1.GB
+  }
+}
+```
+
+The `queue` in this case comes from the Tower UI. First navigate to the workspace in
+question and go to the `Compute Environments` tab. Select the environment in question.
+
+![find-queue-id-1](./find-queue-id-1.png)
+
+Scroll down until you find the ID for `Compute queue`. This is that ID you'll use.
+
+![find-queue-id-2](./find-queue-id-2.png)
+
+<br>
+
+##### Additional example:
+
+The `maxSubmitAwait` directives allows you to specify how long a task can remain in submission queue without being executed. Elapsed this time the task execution will fail.
```

Comment:
* nit: directive*
* _"After this time has elapsed the task execution will fail."_